metomi/rose/date.py

```diff
@@ -482,5 +486,47 @@ def upgrade_offset(offset: str) -> str:
     return result
 
 
+def upgrade_cylc5_datetime(datetime: str) -> str:
+    """Replace a Cylc 5 style (%Y%m%d%h) datetime with a ISO8601 datetime.
+
+    Examples:
+        >>> upgrade_cylc5_datetime('2022010101')
+        '20220101T01'
+    """
+    upgraded = f'{datetime[:-2]}T{datetime[-2:]}'
+
+    print(
+        f'[WARN] This datetime syntax {datetime}'
+        f' is deprecated: Using {upgraded}',
```

Comment:
Maybe indicate that the user should take action?
```suggestion
        f' is deprecated. Use {upgraded} instead',
```